device-selection-explainer.md

```diff
@@ -51,6 +51,29 @@ Later the need for explicit device selection support was challenged in [[MLConte
 
 ## Key use cases and requirements
 
+### Device Preference Use Cases
+
```

Comment:
It would be good to mention this list is based on an API, [OrtExecutionProviderDevicePolicy](https://onnxruntime.ai/docs/api/c/group___global.html#gaf26ca954c79d297a31a66187dd1b4e24), and the text needs to be amplified/checked with the respective use cases.